sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -231,6 +234,21 @@ CosmosClientBuilder setApiType(ApiType apiType){
         return this;
     }
 
+    /**
+     * Adds a policy for modifying request options dynamically.
+     *
+     * @param policy the policy to add
+     * @return current cosmosClientBuilder
+     */
+    public CosmosClientBuilder addOperationPolicy(CosmosOperationPolicy policy) {
+        this.requestPolicies.add(policy);
+        return this;
+    }
+
+    List<CosmosOperationPolicy> getOperationPolicies() {
+        return requestPolicies;
```

Comment:
Do we want to use defensive copying / return an immutable copy - this will force downstream application to use `addOperationPolicy`.

Reply:
Changed to return an immutable copy